### PR TITLE
Quarkiverse extensions can build on other OSes too

### PR DESCRIPTION
**Pull request by gastaldi**
This PR introduces some improvements to Quarkiverse extensions' build script:

- Changes the build script to easily allow building on other OSes (commented out by default) 
- Use actions/setup-java's cache

### CODE REVIEW

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml [hunk 1]
```diff
@@ -24,30 +24,22 @@ on:
 
 jobs:
   build:
-
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml:30]: Why Windows but not osx ?

**Review by maxandersen — APPROVED** [review]
Assuming tests passes lgtm

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml (hunk 1 above)
Comment (gastaldi) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml:30]: Sure

**Review by gsmet — CHANGES_REQUESTED** [review]
Just want to open a debate, please dismiss if you think it's useless.

Is it really a good thing to multiply our CI usage by 3, when in most cases this will have little value? I would think it could be enabled by the Quarkiverse authors when they think it makes sense for their libraries (we had one so far AFAICS).

Also, we wanted to enable native testing by default for Quarkiverse extensions and it will require some non trivial work with Windows in the matrix.
So if we decide to do that, I think we should add native testing on Linux (and I would limit it to Linux for the default behavior) because it might not be easy for the Quarkiverse contributor to do it afterwards with this new setup.

**Comment by gsmet** [comment]
Btw this question: `Is it really a good thing to multiply our CI usage by 3, when in most cases this will have little value?` has 3 dimensions:
- GitHub Actions resources in general
- The planet
- Our GitHub Actions limited resources -> we multiply by 3 our GitHub Actions usage and I know some of you were already worried about the day when we would hit the limit

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml [hunk 2]
```diff
@@ -24,30 +24,22 @@ on:
 
 jobs:
   build:
-
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
+    strategy:
```
Comment (gsmet) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/.github/workflows/build.yml:28]: You need to disable `fail-fast`: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

**Comment by gsmet** [comment]
I wouldn't be against having this matrix setup with Linux by default and a commented line with Linux/Windows.

I see little value in testing with macOS in the Java world, except in very specific cases.

**Comment by maxandersen** [comment]
You are right @gsmet it does not make sense to enable This for All by default. 

Maybe parameteriZe the matrix and let users enable whats needed. 

Osx is relevant for native but yes; not by default.

**Comment by gastaldi** [comment]
@gsmet The reason why I am doing this is because some Quarkiverse builds were reported to fail in Windows (see https://github.com/quarkiverse/quarkus-openapi-generator/issues/7). 

But I agree that it wouldn't be necessary from day one. I like your suggestion of having the matrix setup with Linux by default and some commented out lines for Windows. I also agree that testing in macOS is not very valuable (perhaps if one likes to investigate `too many open files` errors, but that's another story) :)

**Comment by gsmet** [comment]
> some Quarkiverse builds were reported to fail in Windows (see https://github.com/quarkiverse/quarkus-openapi-generator/issues/7).

Yes, I have seen that, that's the one occurrence I was referencing to. I'm just unsure we should generalize this based on a few occurrences where it makes absolutely sense.

CI has a cost, even if it appears free to us, and I think we should try to be good citizens about it when we can.

**Review by gsmet — APPROVED** [review]

**Pull request merged**